src/generic/grid.cpp

```diff
@@ -806,74 +806,181 @@ wxGridCellAttr *wxGridCellAttrData::GetAttr(int row, int col) const
+                                          cellPos + mainSize - editPos);
+                        /*
+                        The above was derived from:
+                        first_del = edit
+                        last_del = min(edit - count - 1, cell + size - 1)
+                        size -= (last_del + 1 - first_del)
+
+                        eliminating the 1's:
+
+                        first_del = edit
+                        last_del = min(edit - count, cell + size)
+                        size -= (last_del - first_del)
+
+                        reducing each by edit:
+
+                        first_del = 0
+                        last_del_plus_1 = min(0 - count, cell + size - edit)
+                        size -= (last_del_plus_1 - 0)
+
+                        after eliminating the 0's and substitution, leaving:
+
+                        size -= min(-count, cell + size - edit)
+
+                        E.g. with a multicell of size 7 and at 2 positions
+                        after the main cell 100 positions are deleted then
+                        the size will not (/can't) be reduced by 100 cells
+                        but by:
+
+                        cellPos + 7 - editPos =       # editPos = cellPos + 2
+                        cellPos + 7 - (cellPos + 2) = # eliminate cellPos
+                        7 - 2 =
+                        5 cells, making the final size 7 - 5 = 2.
+                        */
+
+                    cellAttr->SetSize(isEditingRows ? mainSize : cellRows,
+                                      isEditingRows ? cellCols : mainSize);
                 }
             }
+
+            continue;
         }
-    }
-}
 
-void wxGridCellAttrData::UpdateAttrCols( size_t pos, int numCols )
-{
-    size_t count = m_attrs.GetCount();
-    for ( size_t n = 0; n < count; n++ )
-    {
-        wxGridCellCoords& coords = m_attrs[n].coords;
-        wxCoord col = coords.GetCol();
-        if ( (size_t)col >= pos )
+        if ( editCount < 0 && cellPos < editPos - editCount )
         {
-            if ( numCols > 0 )
-            {
-                // If cols inserted, increment col counter where necessary
-                coords.SetCol(col + numCols);
-            }
-            else if (numCols < 0)
+            // This row/col is deleted and the cell doesn't exist any longer:
+            // Remove the attribute.
+            attrs.RemoveAt(n);
+            n--;
+            count--;
+
+            continue;
+        }
+
+        if ( GetCellSpan(cellRows, cellCols) != wxGrid::CellSpan_Inside )
+        {
+            // Rows/cols inserted or deleted (and this cell still exists):
+            // Adjust cell coords.
+            coords.Set(cellRow + editRowCount, cellCol + editColCount);
+
+            // Nothing more to do: cell is not an inside cell of a multicell.
+            continue;
+        }
+
+        // Handle inside cell's existance, coords, and size.
```

Comment:
```suggestion
        // Handle inside cell's existence, coords, and size.
```

Reply:
Ugh, thanks for spotting. Note to self: looking it up (even the correct one at that as this one looked wrong) is not enough, it also has to be actually changed :).